packages/react-native-web/src/exports/Image/index.js

```diff
@@ -146,6 +147,23 @@ function resolveAssetUri(source): ?string {
   return uri;
 }
 
+function raiseOnErrorEvent(uri, { onError, onLoadEnd }) {
+  if (onError) {
+    onError({
+      nativeEvent: {
+        error: `Failed to load resource ${uri} (404)`
+      }
+    });
+  }
+  if (onLoadEnd) onLoadEnd();
+}
+
+function hasSourceDiff(a: ImageSource, b: ImageSource) {
+  return (
+    a.uri !== b.uri || JSON.stringify(a.headers) !== JSON.stringify(b.headers)
```

Comment:
I noticed you updated this in #13 to:

```suggestion
  if (a.uri !== b.uri) return true;

  const headersA = a.headers || {};
  const headersB = b.headers || {};

  return JSON.stringify(headersA) !== JSON.stringify(headersB);
```

Reply:
Aah this is because in #13, `headers` may or may not exist